aws/resource_aws_sns_topic_subscription.go

```diff
@@ -127,34 +134,27 @@ func resourceAwsSnsTopicSubscriptionUpdate(d *schema.ResourceData, meta interfac
 			attrValue = "true"
 		}
 
-		req := &sns.SetSubscriptionAttributesInput{
-			SubscriptionArn: aws.String(d.Id()),
-			AttributeName:   aws.String("RawMessageDelivery"),
-			AttributeValue:  aws.String(attrValue),
-		}
-		_, err := snsconn.SetSubscriptionAttributes(req)
-
-		if err != nil {
-			return fmt.Errorf("Unable to set raw message delivery attribute on subscription")
+		if err := snsSubscriptionAttributeUpdate(snsconn, d.Id(), "raw_message_delivery", attrValue); err != nil {
+			return err
 		}
 	}
 
 	if d.HasChange("filter_policy") {
-		_, n := d.GetChange("filter_policy")
```

Comment:
FYI: `n` here is a specific naming convention as `d.GetChange()` returns `old, new` 😄  (also makes it easier to duplicate the logic if necessary).

Reply:
makes sense :) I'll change it